### ISSUE TITLE
KeeWeb Window Does not Open When Using Global Auto-Type Hotkey, DB is Locked, and Window is Hidden

### ISSUE THREAD

**Issue by deiden26** [keeweb/keeweb#1179]
**Describe the bug**
The KeeWeb window doesn't open when using the "auto-type (when the app is in the background)" hotkey, (<CTL>-<ATL>-T) under the following circumstances

- Database is locked
- Window is hidden

The global auto-type hotkey works as expected in the following situations

_Database is Locked, Window is Open (but not focussed)_
The KeeWeb window takes the focus, auto-focuses the database password field specifically, and hides upon the successful entry of a password. Finally, the auto-typing occurs after the window closes

_Database is Unlocked, Window is Hidden (but app is running)_
The auto-typing occurs immediately.

**To Reproduce**
1. Open KeeWeb
2. Close the KeeWeb window before authenticating, hiding the window and leaving KeeWeb running in the background
3. Use the global auto-type hotkey in a browser window
4. Nothing happens

**Expected behavior**
The KeeWeb window should take the focus, auto-focus the database password field specifically, and hide upon the successful entry of a password. Finally, the auto-typing should occur after the window closes

**Environment**
![Screen Shot 2019-05-16 at 12 27 42 PM](https://user-images.githubusercontent.com/2993523/57881579-859ee300-77d6-11e9-991f-1ab492f7a430.png)

**Logs**
Just the startup logging. No recognition of the hotkey.
```
index.html:12 2019-05-16T19:28:50.988Z [app] Started in 412ms ¯\_(ツ)_/¯
index.html:12 2019-05-16T19:28:50.990Z [updater] Next update check will happen in 17546s
```

**Comment by antelle** [comment]
Thanks for the detailed description!
Just checked it, it doesn't work when you close the window using the red (x) button, right? If it's hidden using ⌘H then it pops up.

**Comment by deiden26** [comment]
You are correct. Hiding instead of closing the window does allow the global hotkey to work.

**Comment by AmorbiaCacoa** [comment]
Thank you for the temporary solution by using cmd+H.
Something that might be related (I can also open a new Issue for this if you prefer):
I did set the KeeWeb Mac App to never auto lock my database, also all the checkmarks below are not selected. It just should never be locked.
However, when I press the red X to close the app the database locks. And of cause afterwards auto-type is not working properly as described by deiden26.

**Comment by antelle** [comment]
The bot was misconfigured, sorry for a wrong notification.

**Comment by emikaadeo-git** [comment]
Hi,
I would like to expand the thread.

- Firstly about hidden window.

In macOS you got the ablility to open any app automatically when you log in on Mac.
The great thing is that you can also hide a login app (if you don’t want an app window to pops up every time you log in)
https://support.apple.com/guide/mac-help/open-items-automatically-when-you-log-in-mh15189/mac
Above is not working with KeeWeb.

**Current behavior**
Even if set to automatically open and hide, KeeWeb window pops up every time you log in to Mac.

**Expected behavior**
KeeWeb should stay hidden in Dock when log in to Mac.

- Secondly about using global Auto-Type hotkey when database is locked and window is hidden/closed.

When trigger global Auto-Type hotkey, KeeWeb window should pop up no matter if you previously close window with red (x) button or hide it with Command+H shortcut.
Why? Because in both scenarios, app is still running in the background on macOS.

**Current behavior**
KeeWeb window not pops up when you trigger global Auto-Type hotkey if you previously close window with red (x) button.

**Expected behavior**
After trigger global Auto-Type hotkey, KeeWeb window should take the focus, auto-focus the database password field (if the database is locked of course), and after the successful entry of a password, auto-type login and password to website in a browser window.

--------------------------
**Environment**
macOS Catalina 10.15 (19A583)
Firefox browser
KeeWeb v1.11.9

**Comment by emikaadeo-git** [comment]
@antelle any feedback will be highly appreciated ;)

**Comment by antelle** [comment]
@emikaadeo-git I think my opinion is quite clear from tags 😉 however here's what I think:

1. it's a bug that should be fixed and all you said indeed makes sense
2. there's no capacity to fix in the next release, so it's triaged to v1.x
3. the fix is not that easy as one might think
4. regarding launching in dock: launch in-tray option is covered in #480, however it won't solve this; this feature should be handled separately, macOS provides a way to do this, but the support needs to be implemented
